### ISSUE TITLE
Unexpected texels in Zbuffer 

### ISSUE THREAD

**Issue by DuaneNielsen** [facebookresearch/pytorch3d#471]
## 🐛 Bugs / Unexpected behaviors
I think the zbuffer contains color from faces that shouldn't exist.

## Instructions To Reproduce the Issue:

Took the current master branch, compiled it under CUDA 11 on linux and ran it.  See output below...

The below code renders three planar meshes, of different colors, on top of each other in the z axis.

After rasterization, the zbuffer contains the 3 objects/faces, but 2 of the faces contain artifacts, and the empty layers of the zbuffer also contain artifacts, from the edges of the faces.

Increasing blur appears to make things worse.  (Test it and you will see).

```
from pytorch3d.renderer import \
    (TexturesVertex, MeshRenderer, MeshRasterizer, FoVOrthographicCameras, RasterizationSettings,
     look_at_view_transform)
from pytorch3d.structures import Meshes, join_meshes_as_scene
import torch
import torch.nn as nn
import matplotlib.pylab as plt

device = 'cuda' if torch.cuda.is_available() else 'cpu'

""" a square mesh """

scale = 27.0

verts = torch.tensor([
    [-1, -1, 0],
    [1, -1, 0],
    [1, 1, 0],
    [-1, 1, 0]
], dtype=torch.float) * scale

faces = torch.LongTensor([
    [0, 1, 2], [0, 2, 3]
])

white = torch.ones_like(verts)
red = white * torch.tensor([1.0, 0.0, 0.0])
green = white * torch.tensor([0.0, 1.0, 0.0])
blue = white * torch.tensor([0.0, 0.0, 1.0])


def height(z):
    v = verts.clone()
    v[:, 2] = torch.full((4, ), fill_value=z)
    return v


meshes = Meshes(
    verts=[height(1.0), height(0.0), height(-1.0)],
    faces=[faces, faces, faces],
    textures=TexturesVertex([red, green, blue])
)


scene = join_meshes_as_scene(meshes)


class IdentityShader(nn.Module):
    """
    shader that simply returns the raw texels
    """

    def __init__(self):
        super().__init__()

    def forward(self, fragments, meshes, **kwargs) -> torch.Tensor:
        texels = meshes.sample_textures(fragments)
        return texels


cameras = FoVOrthographicCameras(device=device,
                                 max_x=80.0, max_y=93.0,
                                 min_x=-80.0, min_y=-93.0,
                                 scale_xyz=((1, 1, 1),))

raster_settings = RasterizationSettings(
    image_size=128,
    blur_radius=1e-6,
    faces_per_pixel=6,
)


zbuffer_renderer = MeshRenderer(
    rasterizer=MeshRasterizer(
        cameras=cameras,
        raster_settings=raster_settings,
    ),
    shader=IdentityShader()
)

distance, elevation, azimuth = 30, 0.0, 0
R, T = look_at_view_transform(distance, elevation, azimuth, device=device)
zbuffer = zbuffer_renderer(meshes_world=scene.to(device), R=R, T=T)


fig = plt.figure(figsize=(10, 10))
for i in range(1, 7):
    panel = fig.add_subplot(2, 3, i)
    panel.imshow(zbuffer[..., i-1, :].squeeze().cpu())
plt.grid(False)
plt.show()

```

![image](https://user-images.githubusercontent.com/11070508/100831503-c3944400-341a-11eb-8dfa-81973800970b.png)

**Comment by DuaneNielsen** [comment]
Hang on, I think i get it.   That's what blur is?  Pushing pixels to lower layers in the zbuffer around the edges?
